test/e2e/olm_upgrade_test.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"os"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	rayv1alpha1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	. "github.com/project-codeflare/codeflare-operator/test/support"
+)
+
+var (
+	namespace          *corev1.Namespace
+	namespaceName      string = "test-ns-olmupgrade"
+	serviceAccountName string = "sdk-user"
+)
+
+// Creates a Ray cluster
+func TestMNISTRayClusterUp(t *testing.T) {
+
+	test := With(t)
+	test.T().Parallel()
+	if os.Getenv("RUN_OLM_TESTS") != "true" {
+		test.T().Skip("Skipping OLM upgarde test because RUN_OLM_TESTS is not set")
+	}
+	// Create a namespace
+	namespace = CreateTestNamespaceWithName(test, namespaceName)
+	test.T().Logf("Created namespace %s successfully", namespace.Name)
+
+	// Delete namespace only if test failed
+	defer func() {
+		if t.Failed() {
+			DeleteTestNamespace(test, namespaceName)
+		}
+	}()
+
+	// Test configuration
+	config := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mnist-raycluster-sdk",
+			Namespace: namespace.Name,
+		},
+		BinaryData: map[string][]byte{
+			// SDK script
+			"raycluster_sdk.py": ReadFile(test, "raycluster_sdk.py"),
```

Comment:
The file name is not much descriptive.
Thinking about something like `start_ray_cluster.py`. What do you think?

Reply:
yes sounds good